### PR TITLE
test-extended-master-secret-extension.py: advertize curves on ECDHE

### PR DESCRIPTION
**Pull request by nmav**
That is, in test 'extended master secret w/ECDHE'  when attempting
to negotiate ECDHE advertize at least one curve. This makes the test
work with gnutls which will not negotiate ECDHE otherwise.

### CODE REVIEW

File: scripts/test-extended-master-secret-extension.py [hunk 1]
```diff
@@ -194,7 +195,8 @@ def main():
     node = node.add_child(ClientHelloGenerator(
         ciphers,
         extensions={ExtensionType.renegotiation_info:None,
-                    ExtensionType.extended_master_secret:None}))
+                    ExtensionType.extended_master_secret:None,
+                    ExtensionType.supported_groups:SupportedGroupsExtension().create([GroupName.secp256r1])}))
```
Comment (tomato42) [inline scripts/test-extended-master-secret-extension.py]: Line too long

**Comment by tomato42** [comment]
In general, we want tests that check if the ECDHE can be negotiated without client advertising curves - it is stated explicitly as valid in RFC 4492. That being said, it's a test for EMS, not ECDHE, so I guess we can be more explicit about it in this case.

Problem is that the `test-ecdhe-rsa-key-exchange.py` is very simple and does not test the case without curves being advertised, so as this looks to me like the only test case that does explicit ECDHE connection without advertising curves, we need to update the other script first before merging this change.

**Comment by nmav** [comment]
Updated based on the comments and the discussion we had.

**Comment by nmav** [comment]
(note that I copied-pasted code as I don't know well python)
I've tested it against gnutls master branch

**Pull request merged**